api/client/save.go

```diff
@@ -44,14 +45,16 @@ func (cli *DockerCli) CmdSave(args ...string) error {
 		if _, err := cli.stream("GET", "/images/"+image+"/get", sopts); err != nil {
 			return err
 		}
-	} else {
-		v := url.Values{}
-		for _, arg := range cmd.Args() {
-			v.Add("names", arg)
-		}
-		if _, err := cli.stream("GET", "/images/get?"+v.Encode(), sopts); err != nil {
-			return err
-		}
+		return nil
 	}
+
+	v := url.Values{}
+	for _, arg := range cmd.Args() {
```

Comment:
It seems like we can get rid of the whole check before this block (for Args == 1) and just do this.
The `/images/get` and `/images/{name}/get` are the same handlers, I can't imagine them being any different at some point... honestly a very strange API.